server/routes/aws/onboarding.py

```
@@ -304,23 +304,27 @@ def workspace_cleanup(user_id, workspace_id):
 
-        account_id = aws_conn.get('account_id')
-        if account_id:
-            success = delete_connection_secret(user_id, "aws", account_id)
-            if not success:
-                logger.error("Failed to delete AWS connection for user %s account %s", user_id, account_id)
-                return jsonify({"error": "Failed to disconnect AWS connection"}), 500
+        failed_accounts = []
+        for aws_conn in aws_conns:
+            account_id = aws_conn.get('account_id')
+            if account_id:
+                if not delete_connection_secret(user_id, "aws", account_id):
```

Ruff (0.15.14)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)